src/DataTypes/Serializations/ISerialization.cpp

```diff
@@ -288,7 +345,9 @@ bool ISerialization::hasSubcolumnForPath(const SubstreamPath & path, size_t pref
     size_t last_elem = prefix_len - 1;
     return path[last_elem].type == Substream::NullMap
             || path[last_elem].type == Substream::TupleElement
-            || path[last_elem].type == Substream::ArraySizes;
+            || path[last_elem].type == Substream::ArraySizes
+            || path[last_elem].type == Substream::VariantDiscriminators
+            || path[last_elem].type == Substream::VariantElement;
```

Comment:
Here we specify only what we want to read as a subcolumn. And we don't want to read variant offsets as a subcolumn. Also I think we should remove `VariantDiscriminators` from this list, because right now user can request `v.discr` column, but it's more like an implementation detail and won't tell anything useful for the user.
I am also thinkink of adding one more subcolumn `VariantTypes` that in each row it will contain actual type name stored in this row. Smth like `v.variant_types` and add a new function `variantTypeName(...)` that will do the same. Not sure if it will be useful, but not difficult to implement. But maybe in the next PR.

Reply:
Removed `VariantDiscriminators`, should not break anything